### ISSUE TITLE
Train LoRA for judge model to detect missing tool usage

### ISSUE THREAD

**Issue by arkavo-com** [arkavo-org/arkavo-edge#361]
## Objective

Train a LoRA adapter for Gemma-3 270M to reliably detect when LLMs should have used tools but didn't.

## Background

Currently using Gemma-3 270M base model with prompt engineering to judge responses. A fine-tuned LoRA would be:
- More reliable and deterministic
- Faster inference (simpler prompt)
- Better generalization across tool types
- Cost-free (no API calls)

### Data Generation
- [ ] Create synthetic dataset (1000-2000 examples)
- [ ] Categories: time, search, filesystem, shell, network, database, API calls
- [ ] Positive examples (tool used correctly)
- [ ] Negative examples (should have used tool but didn't)
- [ ] Edge cases (doesn't need tools)

### Training
- [ ] Use unsloth for efficient LoRA training
- [ ] Target: Gemma-3 270M base model
- [ ] LoRA rank: 16-32
- [ ] Training time: < 1 hour on single GPU

### Evaluation
- [ ] Accuracy > 95% on held-out test set
- [ ] Inference latency < 200ms on M-series Mac
- [ ] Model size < 50MB (LoRA adapter only)

## Dataset Example Format

**Input:**
```
USER REQUEST: what time is it?
AI RESPONSE: I cannot tell you the current time as I don't have access to real-time information.
AI TOOL CALLS: [No tool calls]
AVAILABLE TOOLS:
- get_agent_time - Get current system time and date
```

**Expected Output:**
```
VERDICT: FAIL
REASON: SUGGEST: time, clock, datetime
ISSUE: missing_tool_use
```

## Implementation Plan

1. Create `crates/arkavo-judge-training/` crate
2. Generate synthetic data using Gemini API (cost: ~$0.10)
3. Train LoRA using unsloth
4. Evaluate on test set
5. Package LoRA weights with release
6. Update judge.rs to load LoRA if available

## Success Criteria

- [ ] Judge accuracy > 95% on test set
- [ ] Inference time < 200ms
- [ ] Works offline (no internet required)
- [ ] LoRA weights < 50MB
- [ ] Generalizes to new tool types not in training data

## Cost Analysis

- Training: 1 hour GPU time (~$0.50 on cloud, free locally)
- Data generation: ~$0.10 Gemini API calls
- **Total: < $1.00 one-time cost**

vs. Using Gemini for judging:
- ~$0.0001 per judgment
- At 10K requests/day = $1/day = $365/year

**LoRA ROI: Pays for itself after 10 requests**

Related to PR #XXX (judge-based tool detection)
